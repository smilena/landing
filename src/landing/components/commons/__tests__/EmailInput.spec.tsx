import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { EmailInput } from '../EmailInput';

const mockStore = createMockStore([]);

afterEach(cleanup);

describe('Given EmailInput component', () => {
  
  test('should render correctly with initial state', () => {
    const state = {
      userData: {
        email: 'smilenaguevara@gmail.com',
      }
    };
    const store = mockStore(state);
    const { container } = render(
      <Provider store={store}>
        <EmailInput id={'email'} label={'Email'}/>
      </Provider>);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('input-email').getAttribute('value')).toEqual("smilenaguevara@gmail.com");
    expect(screen.getByText('Email')).toBeInTheDocument();
  });

  test('should render correctly without initial state', () => {
    const state = {
      userData: {}
    };
    const store = mockStore(state);
    render(
      <Provider store={store}>
        <EmailInput id={'floor'} label={'Email'}/>
      </Provider>);
    expect(screen.getByTestId('input-email').getAttribute('value')).toEqual('');
    expect(screen.getByText('Email')).toBeInTheDocument();
  });
});
