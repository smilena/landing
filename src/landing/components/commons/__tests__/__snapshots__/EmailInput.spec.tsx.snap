// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given EmailInput component should render correctly with initial state 1`] = `
<div>
  <div
    class="sc-bczRLJ eaUlQc"
  >
    <label
      class="sc-gsnTZi bfxakb"
      for="email"
    >
      Email
    </label>
    <input
      class="sc-dkzDqf dWwYGf"
      data-testid="input-email"
      id="email"
      type="email"
      value="smilenaguevara@gmail.com"
    />
  </div>
</div>
`;
